Documentation/Usage.md

```diff
@@ -146,81 +146,106 @@ following files from the libgit2 system-package are of interest:
     /usr/local/lib/libgit2.dylib      # .so on Linux
     /usr/local/include/git2.h
 
-Swift packages that provide module maps for system libraries are handled
-differently from regular Swift packages.
+**Note:** the system library may be located elsewhere on your system, such as
+`/usr/`, or `/opt/homebrew/` if you're using Homebrew on an Apple Silicon Mac.
+If you're not sure where `Clibgit` is in your system, you can run this command
+to look it up:
 
-Note that the system library may be located elsewhere on your system, such as
-`/usr/` rather than `/usr/local/`.
+    example$ pkg-config --cflags libgit2
+    -I/usr/local/libgit2/1.6.4/include
 
-Create a directory called `Clibgit` next to the `example` directory and
-initialize it as a package that builds a system module:
 
-    example$ cd ..
-    $ mkdir Clibgit
-    $ cd Clibgit
-    Clibgit$ swift package init --type empty
-
-This creates a `Package.swift` file in the directory.
-Edit `Package.swift` and add `pkgConfig` parameter:
+**First, let's define the `target` in the package description**:
 
 ```swift
-// swift-tools-version:5.3
+// swift-tools-version: 5.8
+// The swift-tools-version declares the minimum version of Swift required to build this package.
 
 import PackageDescription
 
 let package = Package(
-    name: "Clibgit",
-    pkgConfig: "libgit2"
+    name: "example",
+    targets: [
+        // systemLibrary is a special type of build target that wraps a system library
+        // in a target that other targets can require as their depencency.
+        .systemLibrary(
+            name: "Clibgit",
+            pkgConfig: "libgit2",
+            providers: [
+                .brew(["libgit2"]),
+                .apt(["libgit2-dev"])
+            ]
+        )
+    ]
 )
+
 ```
 
-The `pkgConfig` parameter helps SwiftPM in figuring out the include and library
-search paths for the system library. Note: If you don't want to use the `pkgConfig`
-parameter you can pass the path of a directory containing the library using the
-`-L` flag in commandline when building your app:
+**Note:** If you don't want to use the `pkgConfig` parameter you can pass the
+path of a directory containing the library using the `-L` flag in command line
+when building your app instead:
 
     example$ swift build -Xlinker -L/usr/local/lib/
 
-Create a `module.modulemap` file so it consists of the following:
+Next, create a directory `Sources/Clibgit` in your `example` project, and
+add a `module.modulemap` to it:
 
     module Clibgit [system] {
       header "/usr/local/include/git2.h"
```

Comment:
there is a somewhat better way to do this, namely to create you own header file, which then include the required one with `<header>` so that no path hardcoding is required.  eg: https://github.com/apple/swift-package-manager/tree/main/Sources/SPMSQLite3

Reply:
Sorry, long flight — back in PST now. 

Ah, that's a TIL for me. So it looks like:
- You can, and probably should, add a header file in Sources/{LibName}/{LibName.h} that just includes {LibName.h} and/or other required headers. 
- The `#include` will resolve with the compiler environment, so it should pick up the headers correctly, and thus you don't have to write the absolute header path in the `module.modulemap`. 

Right?

Reply:
yup afaik. @neonichu to keep me honest here